sw/device/tests/aes_interrupt_encryption_test.c

```diff
@@ -0,0 +1,260 @@
+  } while (!ready);
+
+  // Disable the AES unit to no longer automatically start encryption/decryption
+
+  transaction.manual_operation = kDifAesManualOperationManual;
+
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerDataOutClear));
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerKeyIvDataInClear));
+
+  LOG_INFO(" De-initialized Automatic operation ");
+
+  //  Perform an trial encryption using a different data and key. Message size,
+  //  key size and mode can be chosen arbitrarily.
+
+  dif_aes_t aes2;
+
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes2));
+
+  CHECK_DIF_OK(dif_aes_reset(&aes2));
+
+  // Mask the key. Note that this should not be done manually. Software is
+  // expected to get the key in two shares right from the beginning.
+
+  uint8_t key_share0_t[sizeof(kAesModesKey256)];
+  for (int i = 0; i < sizeof(kAesModesKey256); ++i) {
+    key_share0_t[i] = kAesModesKey256[i] ^ kKeyShare1[i];
+  }
+
+  // "Convert" key share byte arrays to `dif_aes_key_share_t`.
+  dif_aes_key_share_t key_t;
+  memcpy(key_t.share0, key_share0, sizeof(key_t.share0));
+  memcpy(key_t.share1, kKeyShare1, sizeof(key_t.share1));
+
+  dif_aes_iv_t iv2;
+  memcpy(iv2.iv, kAesModesIvCtr, sizeof(iv2.iv));
+
+  dif_aes_transaction_t transaction_trial = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeCtr,
+      .key_len = kDifAesKey256,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPerBlock,
+      .manual_operation = kDifAesManualOperationAuto,
+      .reseed_on_key_change = false,
+      .ctrl_aux_lock = false,
+  };
+
+  CHECK_DIF_OK(dif_aes_start(&aes2, &transaction_trial, &key_t, &iv2));
+
+  // "Convert" plain data byte arrays to `dif_aes_data_t` array.
+  enum {
+    kAesNumBlocks_t2 = 4,
+  };
+
+  dif_aes_data_t plain_text2[kAesNumBlocks_t2];
+  dif_aes_data_t cipher_text2[kAesNumBlocks_t2];
+  memcpy(plain_text2[0].data, kAesModesPlainText, sizeof(kAesModesPlainText));
+
+  CHECK_DIF_OK(dif_aes_process_data(&aes2, plain_text2, cipher_text2,
+                                    (size_t)kAesNumBlocks_t2));
+
+  CHECK_DIF_OK(dif_aes_load_data(&aes2, plain_text2[0]));
+  LOG_INFO("Started AES trial Encrypt");
+  CHECK_DIF_OK(dif_aes_end(&aes2));
+
+  /** Not checking successful encryption here since its a trial transaction
+   *  Trial encryption ends here
+   *  Resume the Original Encryption below
+   */
+
+  // Using the iv read from `iv_encrypt1` for resuming the encryption of
+  // remaining 2 blocks.
+  memcpy(iv.iv, &iv_encrypt1, sizeof(iv.iv));
+  transaction.manual_operation = kDifAesManualOperationAuto;
+
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key, &iv));
+
+  memcpy(plain_text[0].data, kAesModesPlainText_block34,
+         sizeof(kAesModesPlainText_block34));
+
+  CHECK_DIF_OK(dif_aes_get_status(&aes, kDifAesStatusInputReady, &ready));
+  LOG_INFO("Resuming Encrypt with the remaining 2 blocks of data ");
```

Comment:
Nit: ditto extraneous whitespace

Reply:
Fixed this in the updated aes_interrupt_encryption_Test.c code